timemachine/md/smc.py

```diff
@@ -0,0 +1,179 @@
+# adapted from https://github.com/proteneer/timemachine/blob/7d6099b0f5b4a2d0b26c3edc7a91c18f7a526c00/md/experimental/smc.py
+
+from typing import Any, Callable, Dict, List, Tuple
+
+import numpy as np
+from scipy.special import logsumexp
+from typing_extensions import TypeAlias
+
+# type annotations
+Sample: TypeAlias = Any
+Samples: TypeAlias = List[Sample]  # e.g. List[CoordsVelBox]
```

Comment:
Curious to understand the motivation for this alias, vs. just using `List[Samples]` directly. Is the idea to abstract over the container type?

Reply:
Mostly to emphasize that the signatures here are all in terms of collections rather than individual samples.

In the future, we may want to use a specialized container for a collection of samples. (Something like an mdtraj Trajectory object, or for toy systems a raw numpy array, or in local MD maybe a new container specialized to the case where most atoms are held constant and only a small number of atoms are mobile.) There may be optimizations that apply to propagating (or computing the potential energy of) a batch of samples, that are not applicable to individual samples.